incubator/jaeger/templates/agent-ds.yaml

```diff
@@ -55,6 +55,9 @@ spec:
         - name: {{ $key | replace "." "_" | replace "-" "_" | upper | quote }}
           value: {{ $value }}
         {{- end }}
+        {{- if .Values.agent.priorityClassName }}
+        priorityClassName: {{ .Values.agent.priorityClassName }}
```

Comment:
priorityClassName goes under the the .spec.template.spec not .spec.template.spec.containers[]